File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter
def solution(k, tangerine):
    answer = 0
    counter = Counter(tangerine)
    
    sort_li = sorted(counter.items(), reverse =True, key = lambda x : x[1])
    
    for key,value in sort_li:
        if k<=0: break
        k-=value
        answer+=1
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    const map = new Map();
    
    tangerine.forEach(t => {
        if(map.has(t)) map.set(t, map.get(t)+1)
        else map.set(t, 1)
    })
    const sortedMap = [...map].sort((a, b) => b[1] - a[1]);
    
    for(let i=0; i<sortedMap.length; i++){
        if(k<=0) break;
        k-=sortedMap[i][1]
        answer+=1;
    }
        
    return answer;
}